rules/rules-reviewed/camel3/camel2/xml-32-changes.windup.xml

```diff
@@ -0,0 +1,216 @@
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_2.html#_jndiregistry" title="Camel 3 - Migration Guide: Spring Boot Starters" />
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="xml-removed-camel32-00001">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="org.apache.camel" artifactId="camel-blueprint" />
+                    </project>
+                    <dependency groupId="org.apache.camel" artifactId="camel-blueprint"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Camel on Apache Karaf / OSGi has been moved to its own project at: https://github.com/apache/camel-karaf" effort="3" category-id="mandatory" >
+                    <message>Camel on Apache Karaf / OSGi has been moved to its own project at: https://github.com/apache/camel-karaf.   Please change the `org.apache.camel:camel-blueprint` dependency to `org.apache.came.karaf:camel-blueprint`.</message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_2.html#_camel_with_karaf_and_osgi" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="xml-removed-camel32-00002">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="org.apache.camel" artifactId="camel-test-blueprint" />
+                    </project>
+                    <dependency groupId="org.apache.camel" artifactId="camel-blueprint"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Camel on Apache Karaf / OSGi has been moved to its own project at: https://github.com/apache/camel-karaf" effort="3" category-id="mandatory" >
+                    <message>Camel on Apache Karaf / OSGi has been moved to its own project at: https://github.com/apache/camel-karaf.   Please change the `org.apache.camel:camel-test-blueprint` dependency to `org.apache.camel.karaf:camel-test-blueprint`.</message>
+                    <link href="https://camel.apache.org/manual/camel-3x-upgrade-guide-3_2.html#_other_components_involved" title="Camel 3 - Migration Guide: Removed components" />
+                </hint>
+            </perform>
+        </rule>
+        <rule id="xml-removed-camel32-00003">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="org.apache.camel" artifactId="camel-test-karaf" />
+                    </project>
+                    <dependency groupId="org.apache.camel" artifactId="camel-test-karaf" />
+                </or>
+            </when>
+            <perform>
+                <hint title="Camel on Apache Karaf / OSGi has been moved to its own project at: https://github.com/apache/camel-karaf" effort="3" category-id="mandatory" >
```

Comment:
Is this a complex change or is it a matter of switching dependencies?

Reply:
It's a change that changes the future of the karaf-camel usage / contributions / ownership / examples / tests.     I think it warrants a 3.

Reply:
:+1: